### PR TITLE
Disallow free promotion from spawning as a CA slot reward

### LINKED ISSUE #429: Promotion CA slot reward isn't dead
Let's kill it properly this time

### PR DESCRIPTION
**Pull request by cannonfodder1**
The sole exception is our Exhaustive Training CA, the whole point of which is to spawn a free promotion.
Closes #429

### CODE REVIEW

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc [hunk 1]
```diff
@@ -1200,6 +1201,34 @@ static protected function string GetAcademySlotBonusDisplayString (XComGameState
...
+	foreach TemplateManager.IterateTemplates(DataTemplate)
+	{
+		ActionTemplate = X2CovertActionTemplate(DataTemplate);
+
+		if (ActionTemplate != none)
+		{
+			// Must be a for loop because a foreach loop is passed a copy of the array
+			for (index = 0; index < ActionTemplate.Slots.Length; index++)
+			{
+				if (ActionTemplate.Slots[index].StaffSlot == 'CovertActionSoldierStaffSlot'
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: What about `CovertActionRookieStaffSlot`? And any other slot template that custom CAs from other mods may use?

Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: Hmm, does that check even need to be there? As far as I can see, staff slots aren't handled differently so it's fine to just call RemoveItem on the empty rewards array

Comment (Xymanek) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: Yeah, I think removing this check is probably the way to go

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc [hunk 2]
```diff
@@ -1200,6 +1201,33 @@ static protected function string GetAcademySlotBonusDisplayString (XComGameState
...
+	foreach TemplateManager.IterateTemplates(DataTemplate)
+	{
+		ActionTemplate = X2CovertActionTemplate(DataTemplate);
+
+		if (ActionTemplate != none)
+		{
+			// Must be a for loop because a foreach loop is passed a copy of the array
+			for (index = 0; index < ActionTemplate.Slots.Length; index++)
+			{
+				if (default.arrAllowPromotionReward.Find(ActionTemplate.DataName) == INDEX_NONE)
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: Move this check outside of the `for` loop, as there is no reason to check it on every iteration

**Pull request merged**